test/new-e2e/pkg/e2e/kubernetes_dump.go

```diff
@@ -0,0 +1,291 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/user"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/util/pointer"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	awsec2 "github.com/aws/aws-sdk-go-v2/service/ec2"
+	awsec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	awseks "github.com/aws/aws-sdk-go-v2/service/eks"
+	awsekstypes "github.com/aws/aws-sdk-go-v2/service/eks/types"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	kubectlget "k8s.io/kubectl/pkg/cmd/get"
+	kubectlutil "k8s.io/kubectl/pkg/cmd/util"
+)
+
+func dumpKubernetesClusterState(ctx context.Context, name string) string {
+	cfg, err := awsconfig.LoadDefaultConfig(ctx)
+	if err != nil {
+		return "WARNING: Failed to dump cluster state: failed to load AWS config"
+	}
+
+	dumpEKS, errEKS := tryDumpEKSClusterState(ctx, cfg, name)
+	if errEKS == nil {
+		return dumpEKS
+	}
+
+	dumpKind, errKind := tryDumpKindClusterState(ctx, cfg, name)
+	if errKind == nil {
+		return dumpKind
+	}
+
+	return fmt.Sprintf("WARNING: Failed to dump cluster state, tried EKS and Kind dump:\n EKS error: %v\n Kind error: %v", errEKS, errKind)
+}
+func tryDumpEKSClusterState(ctx context.Context, cfg aws.Config, name string) (ret string, err error) {
+	var out strings.Builder
+	defer func() { ret = out.String() }()
+
+	client := awseks.NewFromConfig(cfg)
+
+	clusterDescription, err := client.DescribeCluster(ctx, &awseks.DescribeClusterInput{
+		Name: &name,
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to describe cluster %s: %v", name, err)
```

Comment:
Compared to [the original implementation](https://github.com/DataDog/datadog-agent/blob/1b0841a3a05c8cb5d9ee1f0720e91b86ad7faa9c/test/new-e2e/tests/containers/dump_cluster_state.go), this error and all the other ones raised by this function will never be logged anywhere.
It will be problematic if/when the dump of the cluster state itself will be broken.

Reply:
I think the error should be logged on failure, here: https://github.com/DataDog/datadog-agent/pull/26198/files/95d57d2d3fe7b80e5bc97771f6945d70330fe157#diff-8cb88dddcb90c4f6f00a99e5a8125a562f9956ec138669b9320d4d7bc2ef7198R53.
Since we cannot determine if the Kubernetes cluster is running in EKS or in Kind I need to run both and return the first that succeeded, I return error early to avoid running the whole dump function if we are not trying the correct scenario.

But I catch the errors returned and print them in the logs